medallion/test/test_startup.py

```diff
@@ -0,0 +1,338 @@
+import logging
+from unittest import mock
+
+import pytest
+import pytest_subtests  # noqa: F401
+
+import medallion.config
+import medallion.scripts.run
+
+
+def test_parser_help(capsys):
+    """
+    Confirm that the parser help can be printed with the custom formatter.
+    """
+    parser = medallion.scripts.run._get_argparser()
+    parser.print_help()
+    (out, _) = capsys.readouterr()
+    assert "medallion v" in out
+
+
+def test_config_path(subtests):
+    """
+    Confirm that the positional CONFIG_PATH argument works as expected.
+    """
+    config_path_arg = "/we're/on/the/road/to/nowhere"
+    parser = medallion.scripts.run._get_argparser()
+    with subtests.test(msg="CONFIG_PATH provided"):
+        args = parser.parse_args([config_path_arg])
+        assert args.CONFIG_PATH == config_path_arg
+    with subtests.test(msg="CONFIG_PATH omitted"):
+        args = parser.parse_args([])
+        assert args.CONFIG_PATH is None
+
+
+def test_conffile(subtests):
+    """
+    Confirm that the --conf-file option works as expected.
+    """
+    config_path_arg = "/we're/on/the/road/to/nowhere"
+    parser = medallion.scripts.run._get_argparser()
+    with subtests.test(msg="--conf-file omitted"):
+        args = parser.parse_args([])
+        assert args.conf_file == medallion.config.DEFAULT_CONFFILE
+    with subtests.test(msg="--conf-file with space"):
+        args = parser.parse_args(["--conf-file", config_path_arg])
+        assert args.conf_file == config_path_arg
+    with subtests.test(msg="--conf-file with equals"):
+        args = parser.parse_args([f"--conf-file={config_path_arg}"])
+        assert args.conf_file == config_path_arg
+    with subtests.test(msg="MEDALLION_CONFFILE envvar"):
+        with mock.patch.dict(
+            "os.environ", {"MEDALLION_CONFFILE": config_path_arg}
+        ):
+            # We need to make a new parser after mocking the environment
+            parser_with_envvar = medallion.scripts.run._get_argparser()
+            args = parser_with_envvar.parse_args([])
+        assert args.conf_file == config_path_arg
+
+
+def test_confdir(subtests):
+    """
+    Confirm that the --conf-dir option works as expected.
+    """
+    config_path_arg = "/we're/on/the/road/to/nowhere"
+    parser = medallion.scripts.run._get_argparser()
+    with subtests.test(msg="--conf-dir omitted"):
+        args = parser.parse_args([])
+        assert args.conf_dir == medallion.config.DEFAULT_CONFDIR
```

Comment:
This test seems to fail when environment variables are set on Windows. `args.conf_dir` comes out as the environment variable I set for `MEDALLION_CONFDIR`.

Reply:
Urgh, I made a bunch of patches for `os.environ` but must have forgotten to do that for these ones. TBH I think a module level fixture for the two new test modules would be more sensible so that we don't need to think about it in future changes.

Reply:
Okay, I believe this (and the variants commented below) should be fixed now